### PR TITLE
Fix chart sort and live update

### LINKED ISSUE #1086: Sorting charts doesn't work
# Description

https://user-images.githubusercontent.com/44864521/130286302-5c936304-7d42-4a58-ba68-8f46705fb66a.mov

# Location

- [x] Browser
- [ ] CLI
- [ ] API

# Steps to reproduce

1. Go to Applications / Jobs
2. Try to sort using the given select field.

# Additional Details

### LINKED ISSUE #1090: Chart list only refreshes for updates, not addition / deletion
# Description

Noticed that chart list refreshes for updates, was expecting it to do the same for new chart deployments / deletions.

# Location

- [x] Browser
- [ ] CLI
- [ ] API

# Steps to reproduce

1. Open 2 tabs to the "Applications" list view
2. Create an application in tab 1.
3. Check tab 2.

# Additional Details

### PR DESCRIPTION
**Pull request by anukul**
## Pull request type

Closes #1086
Closes #1090

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Other (please describe):

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] If it's a backend change, tests for the changes have been added and `go test ./...` runs successfully from the root folder.
- [ ] If it's a frontend change, Prettier has been run
- [ ] Docs have been reviewed and added / updated if needed

## What is the current behavior?

See #1086, #1090

## What is the new behavior?

Sorting works, chart list refreshes for additions / deletions.

### CODE REVIEW

File: dashboard/src/main/home/cluster-dashboard/chart/ChartList.tsx [hunk 1]
```diff
@@ -212,6 +192,35 @@ const ChartList: React.FunctionComponent<Props> = ({
     return () => (isSubscribed = false);
   }, [namespace, currentView]);
 
+  useEffect(() => {
```
Comment (jnfrati) [inline dashboard/src/main/home/cluster-dashboard/chart/ChartList.tsx:195]: Probably a useMemo would fit better here

File: dashboard/src/main/home/cluster-dashboard/chart/ChartList.tsx [hunk 2]
```diff
@@ -113,21 +84,30 @@ const ChartList: React.FunctionComponent<Props> = ({
...
-          const currentRelease = oldReleases[object?.name];
-          const currentReleaseVersion = Number(currentRelease?.version);
-          const newReleaseVersion = Number(object?.version);
-          if (currentReleaseVersion > newReleaseVersion) {
-            return {
-              ...oldReleases,
-            };
+        const newChart = event.Object;
+        const matches = (chart: ChartType) => chart.name === newChart.name;
+        setCharts((currentCharts) => {
```
Comment (jnfrati) [inline dashboard/src/main/home/cluster-dashboard/chart/ChartList.tsx:89]: Warning here updating always, we should only update on new versions of the chart, as the websocket will return past versions of some charts too.

Comment (anukul) [reply dashboard/src/main/home/cluster-dashboard/chart/ChartList.tsx:91]: oh, thanks for the heads up. not sure why that's happening 🤔

**Review by jnfrati — CHANGES_REQUESTED** [review]
It looks really good! I would like @abelanger5 to check the backend change but I think it looks nice.

Although I would like to discuss some stuff on the websocket implementation if its not blocking you too much

**Review by abelanger5 — APPROVED** [review]
Backend looks good @anukul!

**Review by jnfrati — APPROVED** [review]
Looks awesome! Let's merge it :+1:

**Pull request merged**